pkg/internal/vault/vault.go

```diff
@@ -142,12 +142,15 @@ func (v *Vault) Sign(csrPEM []byte, duration time.Duration) (cert []byte, ca []b
 		return nil, nil, fmt.Errorf("unable to convert certificate bundle to PEM bundle: %s", err.Error())
 	}
 
-	var caPem []byte = nil
+	var caPem []byte
 	if len(bundle.CAChain) > 0 {
-		caPem = []byte(bundle.CAChain[0])
+		caPem = []byte(bundle.CAChain[len(bundle.CAChain)-1])
 	}
 
-	return []byte(bundle.ToPEMBundle()), caPem, nil
+	crtPems := []string{bundle.Certificate}
+	crtPems = append(crtPems, bundle.CAChain[0:len(bundle.CAChain)-1]...)
```

Comment:
Can `bundle.CAChain` ever have a length of 0? If so, this will cause a panic 🙈

Reply:
no, it should always include a CA certificate of a mounted Vault PKI backend, which Vault issuer points to.

Reply:
Given we have a check above for `len(bundle.CAChain) > 0`, can we remove either that check, or just push this into there too? Seems like it'd be safest to wrap it, as I don't see in the Vault docs an explicit call-out that this field will always be set :) but if there is some concrete evidence that this will always be set, happy to assume it is (and perhaps add a comment noting that we are making this assumption)

Reply:
oops, this is wrong, `ca_chain` might be omitted when no certificate chain is registered. I'll fix this later

Reply:
fixed https://github.com/jetstack/cert-manager/pull/3433/commits/c0fe8643e74cc0f8cee1de56ee9fddcc07fe7633 but 

server may not return ca_chain but client ensures it to be set https://github.com/hashicorp/vault/blob/3298836f6a90d95ff97af33cb2abd1f7269847df/sdk/helper/certutil/types.gol